### PR TITLE
make api logging configurable

### LINKED ISSUE #1132: Structured Logging
## Detailed Description

The current logging isn't structured and therefore hard to parse. Also request logging cannot be disabled. I'm suggesting to add a leveled JSON logger which can be enabled via the config file and which allows to disable request logs via the log level.

Log messages could look like this:
`{"lvl":"debug","ts":"2022-12-09T14:50:19+01:00","msg":"some message"}`

`{"lvl":"warn","remote":"::1","method":"POST","path":"/api/files/hello-world-0.0.1","protocol":"HTTP/1.1","code":"400","latency":"208.625µs","agent":"PostmanRuntime/7.29.2","ts":"2022-12-09T14:50:31+01:00","msg":"Error #01: request Content-Type isn't multipart/form-data"}`

## Context

This could also be extended to audit logging (given that not only requests but also responses are logged) which could be used to maintain an inventory of all packages that were signed by aptly. This inventory could then be matched against a list of packages that should have been signed, and thus, allow to detect anomalies.

### PR DESCRIPTION
**Pull request by mmianl**
Fixes #1132

## Requirements

All new code should be covered with tests, documentation should be updated. CI should pass.

## Checklist

- [x] unit-test added (if change is algorithm)
- [x] functional test added/updated (if change is functional)
- [ ] man page updated (if applicable)
- [ ] bash completion updated (if applicable)
- [x] documentation updated
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by mmianl** [comment]
This adds a json logger which prints all logs in a structured manner. If the logging config is left at defaults, then this PR doesn't change the current logging behavior.

Also, I updated go versions to 1.17, 1.18 and 1.19 in the Actions file.

**Comment by mmianl** [comment]
I extended test coverage to meet the required threshold. Also, this PR fixes an issue in the getBasePath function (https://github.com/aptly-dev/aptly/blob/master/api/middleware.go#L17) used by the metrics middleware where accessing the api at `/` would cause an internal server error due to the getURLSegment function accessing the urlSegments out of its range.

**Comment by mmianl** [comment]
@randombenj do think we can merge this?

**Review by randombenj — COMMENTED** [review]
Thanks for those changes. I deffinitly think it does make sense to have some more structured logging for the api. Couldn't we just replicate the current behaviour also through zerolog (https://github.com/rs/zerolog#pretty-logging) and then we can get rid of the whole abstraction in `utils/logging.go` and use zerolog directly just with different configurations. Let me know what you think.

File: utils/logging.go [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+	if err == nil {
+		zerolog.SetGlobalLevel(level)
+		l.level = level
+	} else {
+		zerolog.SetGlobalLevel(zerolog.DebugLevel)
+		l.level = zerolog.DebugLevel
+	}
+
+	zerolog.MessageFieldName = "msg"
+	zerolog.LevelFieldName = "lvl"
```
[the remark this answers is not in the document]
Comment (mmianl) [reply utils/logging.go:405]: The reason for the use of abbreviations here is to save a few bytes on every emitted log message. But if you prefer, I can of course change this to 'level' and 'message'.

**Comment by randombenj** [comment]
Close in vavour of: https://github.com/aptly-dev/aptly/pull/1147
